base/shared/src/main/scala/scalaz/data/kleisli.scala

```diff
@@ -77,18 +81,30 @@ private[data] object KleisliImpl extends KleisliModule {
   override def first[F[_], A, B, C](
     k: Kleisli[F, A, B]
   )(implicit F: Functor[F]): Kleisli[F, (A, C), (B, C)] =
-    wrapKleisli(t => F.map(runKleisli(k)(t._1))((_, t._2)))
+    t => F.map(k(t._1))((_, t._2))
 
   override def second[F[_], A, B, C](
     k: Kleisli[F, A, B]
   )(implicit F: Functor[F]): Kleisli[F, (C, A), (C, B)] =
-    wrapKleisli(t => F.map(runKleisli(k)(t._2))((t._1, _)))
+    t => F.map(k(t._2))((t._1, _))
+
+  override def dimap[F[_], A, B, C, D](
+    k: Kleisli[F, A, B]
+  )(
+    ca: C => A
+  )(
+    bd: B => D
+  )(implicit F: Functor[F]): Kleisli[F, C, D] =
+    c => F.map(k(ca(c)))(bd)
```

Comment:
should this delegate to the `Profunctor` instance or _vice versa_? not like either is likely to change.

Reply:
Think it should mostly be the instance delegating to the functions.